File: twitter-fetcher/src/test/java/org/jesperancinha/twitter/data/MessageTest.java
package org.jesperancinha.twitter.data;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MessageTest {

    @Test
    public void testEquals_whenEquals_Ok() {
        final MessageDto messageDto1 = MessageDto.builder().id("dsfsfndsivnka324")
                .createdAt(111L)
                .text("wowowow")
                .build();

        final MessageDto messageDto2 = MessageDto.builder().id("dsfsfndsivnka324")
                .createdAt(1234324L)
                .text("nononono")
                .build();

        assertThat(messageDto1).isEqualTo(messageDto2);
    }

    @Test
    public void testHashCode_whenOther_NotSame() {
        final MessageDto messageDto1 = MessageDto.builder().id("dsfsfndsivnka324")
                .createdAt(111L)
                .text("wowowow")
                .build();

        final MessageDto messageDto2 = MessageDto.builder().id("dsfsfndsivnka324")
                .createdAt(1234324L)
                .text("nononono")
                .build();

        assertThat(messageDto1).isNotSameAs(messageDto2);
        assertThat(messageDto1.hashCode()).isEqualTo(messageDto2.hashCode());

    }
}

File: news-cast-explorer-fetcher/src/test/kotlin/org/jesperancinha/newscast/data/MessageTest.kt
package org.jesperancinha.newscast.data

import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldNotBeSameInstanceAs
import org.junit.jupiter.api.Test

class MessageTest {
    @Test
    fun testEquals_whenEquals_Ok() {
        val messageDto1 = MessageDto.builder().newsCastId("dsfsfndsivnka324")
            .createdAt(111L)
            .text("wowowow")
            .build()
        val messageDto2 = MessageDto.builder().newsCastId("dsfsfndsivnka324")
            .createdAt(1234324L)
            .text("nononono")
            .build()
        messageDto1 shouldBe messageDto2
    }

    @Test
    fun testHashCode_whenOther_NotSame() {
        val messageDto1 = MessageDto.builder().newsCastId("dsfsfndsivnka324")
            .createdAt(111L)
            .text("wowowow")
            .build()
        val messageDto2 = MessageDto.builder().newsCastId("dsfsfndsivnka324")
            .createdAt(1234324L)
            .text("nononono")
            .build()
        messageDto1 shouldNotBeSameInstanceAs messageDto2
        messageDto1.hashCode() shouldBe messageDto2.hashCode()
    }
}